### PR TITLE
Generate low-S DER signatures

### PR DESCRIPTION
**Pull request by rnicoll**
This ensures all signatures have the lowest possible S value. Where S values are above half the order of the curve, the order of the curve is subtracted from the S value in order to reduce it.

Additional tests for signing added as part of this process.

Low-S value checks have been added to script.py as they are part of script code in Bitcoin Core, but may make more sense in key.py, let me know what you think?

### CODE REVIEW

File: bitcoin/core/key.py [hunk 1]
```diff
@@ -21,6 +21,8 @@
 import hashlib
 import sys
 
+import script
```
Comment (petertodd) [inline bitcoin/core/key.py]: This should be changed to a non-relative import; fails the python3 unit tests.

**Comment by petertodd** [comment]
re: key.py or script.py, I'm inclined to follow Bitcoin Core here and keep the check in script.py

**Comment by rnicoll** [comment]
Changed imports

**Comment by petertodd** [comment]
Fails the python3 unittests:

======================================================================
ERROR: test_high_s_value (bitcoin.tests.test_script.Test_IsLowDERSignature)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pete/src/bitcoin/python-bitcoinlib/bitcoin/tests/test_script.py", line 433, in test_high_s_value
    self.assertFalse(IsLowDERSignature(sig))
  File "/home/pete/src/bitcoin/python-bitcoinlib/bitcoin/core/script.py", line 808, in IsLowDERSignature
    length_r = struct.unpack('B', sig[3])[0]
TypeError: 'int' does not support the buffer interface

**Comment by rnicoll** [comment]
Actually tested with Python 3 this time, sorry about that!

**Pull request merged**